logs-shipper/src/main/java/com/Resource.java

```diff
@@ -43,4 +64,81 @@ public String sendLog() {
         return response;
     }
 
+    @POST
+    @Path("index")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response createTrackInJSON(RequestIndexMessage requestIndexMessage, @HeaderParam("user-agent") String userAgent) {
+        IndexRequest request = new IndexRequest("posts", "_doc");
+        request.id("" + ID++);
+
+        String msg = requestIndexMessage.getMessage();
+        if (msg == null) {
+            return Response.status(HttpURLConnection.HTTP_BAD_REQUEST).entity("The given object is null").build();
+        }
+
+        Map<String, String> jsonAsMap = new HashMap<>();
+        jsonAsMap.put("message", msg);
+        jsonAsMap.put("User-Agent", userAgent);
+
+        request.source(jsonAsMap);
+
+        client.indexAsync(request, RequestOptions.DEFAULT, new ActionListener<IndexResponse>() {
+            @Override
+            public void onResponse(IndexResponse indexResponse) {
+                System.out.println("Success Index");
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                System.out.println("Failure Index");
+                e.printStackTrace();
+            }
+        });
+
+        String result = "Success";
+        return Response.status(HttpURLConnection.HTTP_OK).entity(result).build();
+    }
```

Comment:
Let's try to extract methods
I think the code will be more clear if it will look like:
```java
Map<String, Object> docToIndex = buildDocToIndex(requestIndexMessage, userAgent);
IndexResponse indexResponse = indexToElasticsearch(docToIndex);
return buildResponse(indexResponse);
```

Reply:
done something "similar"